numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -2331,6 +2381,39 @@ npyiter_ass_subscript(NewNpyArrayIterObject *self, PyObject *op,
     return -1;
 }
 
+static PyObject *
+npyiter_self(NewNpyArrayIterObject *self)
+{
+    if (self->iter == NULL)
+    {
+        PyErr_SetString(PyExc_ValueError, "operation on  non-initialized iterator");
+        return NULL;
+    }
+    self->managed &= ITR_INSIDE;
+    Py_INCREF(self);
+    return (PyObject *)self;
+}
+
+static PyObject *
+npyiter_exit(PyObject *f, PyObject *args)
```

Comment:
Why not declare `f` as `NewNpyArrayIterObject` like you do in `npy_iter_self`?

Reply:
fixed in 9a45b3d055